includes/api/ApiQueryUserContributions.php

```diff
@@ -115,23 +91,29 @@ public function execute() {
 	}
 
 	/**
-	 * Validate the 'user' parameter and set the value to compare
-	 * against `revision`.`rev_user_text`
+	 * Validate the 'user' parameter
+	 * and save user name in $this->username,
+	 * $this->userId and $this->userIp
 	 *
 	 * @param $user string
 	 */
 	private function prepareUsername( $user ) {
 		if ( !is_null( $user ) && $user !== '' ) {
-			$name = User::isIP( $user )
-				? $user
-				: User::getCanonicalName( $user, 'valid' );
-			if ( $name === false ) {
-				$this->dieUsage( "User name {$user} is not valid", 'param_user' );
+			// anon
+			if ( User::isIP( $user ) ) {
+				$this->userIp = $user;
+				$this->username = $user;
 			} else {
-				$this->usernames[] = $name;
+				$name = User::getCanonicalName( $user, 'valid' );
+
+				if ( $name === false ) {
+					$this->dieUsage( "User name {$user} is not valid", 'param_user' );
+				} else {
+					// logged in user
+					$this->username = $name;
+					$this->userId = $id = User::idFromName( $name );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$id" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)